homeassistant/components/lifx/button.py

```diff
@@ -0,0 +1,105 @@
+"""Button entity for LIFX devices.."""
+from __future__ import annotations
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, IDENTIFY, RESTART
+from .coordinator import LIFXUpdateCoordinator
+
+RESTART_BUTTON_DESCRIPTION = ButtonEntityDescription(
+    key=RESTART,
+    name="Restart",
+    device_class=ButtonDeviceClass.RESTART,
+    entity_registry_enabled_default=False,
+    entity_category=EntityCategory.DIAGNOSTIC,
+)
+
+IDENTIFY_BUTTON_DESCRIPTION = ButtonEntityDescription(
+    key=IDENTIFY,
+    name="Identify",
+    entity_registry_enabled_default=False,
+    entity_category=EntityCategory.DIAGNOSTIC,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up LIFX from a config entry."""
+    domain_data = hass.data[DOMAIN]
+    coordinator: LIFXUpdateCoordinator = domain_data[entry.entry_id]
+    async_add_entities(
+        [
+            LIFXRestartButton(
+                coordinator=coordinator, description=RESTART_BUTTON_DESCRIPTION
+            ),
+            LIFXIdentifyButton(
+                coordinator=coordinator, description=IDENTIFY_BUTTON_DESCRIPTION
+            ),
+        ]
+    )
+
+
+class LIFXButton(CoordinatorEntity[LIFXUpdateCoordinator], ButtonEntity):
```

Comment:
Let's pull out the duplicate code between `light.py` and `button.py` entity class into an `entity.py` so we have a base class for lifx entities that sets `_attr_device_info`